specification.md

```diff
@@ -2571,6 +2572,30 @@ peer and adjust their behavior appropriately. How exactly the behavior is
 adjusted for future capabilities MUST be defined in the future specification of
 the new capabilities.
 
+### Protobuf Schema Stability
+
+The specification provides the follow stability guarantees of the
+[Protobuf definitions](proto/opamp.proto) for OpAMP 1.0:
+
+- Field types, numbers and names will not change.
+- Names of messages and enums will not change.
+- Names of enum choices and numbers assigned to enum choices will not change.
+- The location of messages and enums, i.e. whether they are declared at the top lexical
+  scope or nested inside another message will not change.
+- Package names and directory structure will not change.
+- Existing symbols will not be deleted.
+- `optional` and `repeated` declarators of existing fields will not change.
```

Comment:
Theoretically, since adding new fields is allowed in the future, perhaps we should not guarantee preservation of `optional` declarators.

Reply:
The reason we want to preserve the `optional` declarator is because changing can change the generated code (see [Protobuf presence API](https://github.com/protocolbuffers/protobuf/blob/main/docs/field_presence.md#presence-in-proto3-apis)). This can then result in breaking the OpAMP language library's API if the generated code is part of the API (which it is in opamp-go, for example).

Reply:
Got it. Dropping `optional` results in dropping hazzers (in proto3) in the generated code.
BTW, I noticed that the current proto files for OpAMP do not use the `optional` label, but sometimes mention optionality in comments. Has this been done on purpose?

Reply:
> BTW, I noticed that the current proto files for OpAMP do not use the `optional` label, but sometimes mention optionality in comments. Has this been done on purpose?

The support of presence functions for `optional` fields was added relatively recently and the support in Protobuf generators was not very good last time I checked (some languages support it some others don't), so it was safer not to use `optional` and rely on comments at the time. The situation may be different now.

You are right though that we don't use `optional` currently and _adding_ it to any field should be backward compatible (_removing_ is not backward compatible though). I think it is probably not a bad idea to keep this requirement, otherwise _optional_ may be added to a field and then remove in some other future version and end up breaking code.

This is probably stronger guarantee than we need, but it is benign since it doesn't restrict us in any way.